src/Microsoft.Framework.Logging.Testing/LogValuesAssert.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright(c) Microsoft Open Technologies, Inc.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit.Sdk;
+
+namespace Microsoft.Framework.Logging.Testing
+{
+    public static class LogValuesAssert
+    {
+        /// <summary>
+        /// Asserts that all the expected values are present in the actual values by ignoring
+        /// the order of values.
+        /// </summary>
+        /// <param name="expectedValues">Expected subset of values</param>
+        /// <param name="actualValues">Actual set of values</param>
+        /// <param name="comparer">
+        /// An <see cref="IEqualityComparer{T}"/> to compare the values. 
+        /// In case no comparer is provided, a default comparer is used.
+        /// </param>
+        public static void Contains(
+            IEnumerable<KeyValuePair<string, object>> expectedValues,
+            IEnumerable<KeyValuePair<string, object>> actualValues,
+            IEqualityComparer<KeyValuePair<string, object>> comparer = null)
+        {
+            if(expectedValues == null && actualValues == null)
+            {
+                return;
+            }
+
+            if(expectedValues == null || actualValues == null)
+            {
+                throw new EqualException(
+                        expected: GetString(expectedValues),
+                        actual: GetString(actualValues));
+            }
+
+            if (comparer == null)
+            {
+                comparer = new LogValueComparer();
+            }
+
+            foreach (var expectedPair in expectedValues)
+            {
+                bool isItemFound = false;
+                foreach (var actualPair in actualValues)
```

Comment:
Can this loop just be `bool isItemFound = actualValues.Contains(expectedPair, comparer);`?